tests/Unit/controllers/account.test.js

```diff
@@ -0,0 +1,494 @@
+        };
+      });
+
+      accountRepository.generateAuthToken = jest.fn((account) => expectedToken);
+
+      relationshipsRepository.create = jest.fn((account) => {});
+
+      groupsRepository.create = jest.fn((account) => {});
+
+      await controller.create(req, res, next);
+
+      expect(accountRepository.create).toHaveBeenCalledTimes(1);
+      expect(accountRepository.create).toBeCalledWith({
+        newAccount: expectedAccount,
+      });
+      expect(relationshipsRepository.create).toHaveBeenCalledTimes(1);
+      expect(relationshipsRepository.create).toBeCalledWith(expectedAccountId);
+      expect(groupsRepository.create).toHaveBeenCalledTimes(1);
+      expect(groupsRepository.create).toBeCalledWith(expectedAccountId);
+      expect(accountRepository.generateAuthToken).toHaveBeenCalledTimes(1);
+      expect(accountRepository.generateAuthToken).toBeCalledWith(
+        expectedAccount
+      );
+      expect(res.status).toHaveBeenCalledTimes(1);
+      expect(res.status).toBeCalledWith(httpStatus.CREATED);
+      expect(res.json).toHaveBeenCalledTimes(1);
+      expect(res.json).toBeCalledWith({ token: expectedToken });
+      expect(next).toHaveBeenCalledTimes(0);
+    });
+
+    test("Should throw invalid object error when request contains invalid falids and handle it", async () => {
+      //This test will fail until we redo validation
+      const expectedError = new Error(errors.INVALID_OBJECT);
+      expectedError.name = errors.INVALID_OBJECT;
+
+      const req = { body: {} };
+      const res = {};
+      const next = jest.fn();
+
+      await controller.create(req, res, next);
+
+      expect(next).toHaveBeenCalledTimes(1);
+      expect(next).toBeCalledWith(expectedError);
+    });
+
+    test("Should handle errors", async () => {
+      const expectedError = new Error(errors.UNKNOWN);
+      expectedError.name = errors.UNKNOWN;
+
+      accountRepository.create = jest.fn((opts) => {
+        throw expectedError;
+      });
+
+      const req = { body: {} };
+      const res = {};
+      const next = jest.fn();
+
+      await controller.create(req, res, next);
+
+      expect(next).toHaveBeenCalledTimes(1);
+      expect(next).toBeCalledWith(expectedError);
+    });
+  });
+
+  describe("Update account", () => {
+    test("Should update successfully when given valid updates", async () => {
+      const expectedUpdates = {
+        userName: "New user name",
+        email: "New email",
+        password: "New password",
+      };
+
+      const req = {
+        body: {
+          ...expectedUpdates,
+        },
+        auth: {},
+      };
+      const res = {
+        send: jest.fn(),
+      };
+      const next = jest.fn();
+
+      accountRepository.update = jest.fn((opts) => {
+        expect(opts.updates).toEqual(expectedUpdates);
+      });
+
+      await controller.update(req, res, next);
+
+      expect(accountRepository.update).toHaveBeenCalledTimes(1);
+      expect(res.send).toHaveBeenCalledTimes(1);
```

Comment:
its better to call toHaveBeenCalledWith(expectedUpdates) for both of these expects, since these line would just require the function to be called but not necessarily with correct values